bleep-core/src/scala/bleep/GenBloopFiles.scala

```diff
@@ -328,6 +331,52 @@ object GenBloopFiles {
         case _                                            => bloop.config.Tag.Library
       }
 
+    val sourceGenerators = {
+      val sg = explodedProject.`source-generator-scripts`.values.map { scriptName =>
+        if (!build.scripts.contains(scriptName)) {
+          sys.error(s"Unknown script ${scriptName} referenced by ${crossName.value}")
+        }
+        val codegenPath = buildPaths.generatedSourcesDir(crossName)
+        val bleepPath = bleepExecutable.getPath().toString
+        val scriptCall = List(bleepPath, scriptName.value, "--project", crossName.value)
+        // Transitive source-path of the script, used as a way to communicate to
+        // bloop what to watch to invalidate its code-generation cache.
+        // This implies that any code-generator input (say, .proto files),
+        // ought to be part of the resources of the script (or of its parents')
+        val scriptSources: List[Config.SourcesGlobs] =
+          build
+            .scripts(scriptName)
+            .values
+            .flatMap {
+              case model.ScriptDef.Main(scriptProjectName, _, _) =>
+                val withTransitive = List(scriptProjectName) ++ build.transitiveDependenciesFor(scriptProjectName).keys
+                withTransitive.flatMap { projectName =>
+                  val bloopProject = getBloopProject(projectName).project
+                  bloopProject.sources ++ bloopProject.resources.getOrElse(Nil)
+                }
+              case model.ScriptDef.Shell(_, _) =>
+                throw new BleepException.Text(s"cannot use shell script as resource generator, because bleep doesn't know how to invalidate it")
```

Comment:
we can use shell scripts now if we add `sourceGlobs` to `ScriptDef.Shell` and assert here that it's non-empty